### PR TITLE
Programming Languages - Additional Resources for "TypeScript"

### PR DESCRIPTION
**Pull request by MastanSayyad**
**IMPORTANT:** When creating your Pull Request, please use the following format for the title: `Category - Resource Name & Section`  

### Description: 
Hey @jfmartinz I have added some additional resources for Typescript such as
- Typescript Validation
- Additional TS Libraries
- Additional Books

### Checklist:
Before submitting your pull request, ensure that you have completed the following tasks: 

- [X] I have carefully reviewed and adhered to the [contributing guidelines](https://github.com/jfmartinz/ResourceHub/blob/main/CONTRIBUTING.md) before creating this pull request.
- [X] I followed the prescribed PR title template. _(Check this if you're adding a resource)_

### CODE REVIEW

File: Programming Languages/TypeScript/readme.md [hunk 1]
```diff
@@ -736,6 +737,151 @@ Below are some essential resources to help you get started and deepen your under
...
+
+
+### TypeScript Validation
+
+> Validate your TypeScript data and ensure type safety with these validation libraries:
+
+<table>
+  <thead>
+    <tr>
+      <th>Library</th>
```
Comment (jfmartinz) [inline Programming Languages/TypeScript/readme.md:825]:
```suggestion
      <th>Resource Name</th>
```

For consistency, please rename it, thanks 👍

**Review by jfmartinz — CHANGES_REQUESTED** [review]

**Comment by MastanSayyad** [comment]
Done @jfmartinz , I'll make sure it does not happen next time

**Comment by jfmartinz** [comment]
> Done @jfmartinz , I'll make sure it does not happen next time

No worries @MastanSayyad

**Review by jfmartinz — APPROVED** [review]
Thanks

**Pull request merged**